### PR TITLE
RedCanary - redcanary-get-detection is causing timeout

### PR DESCRIPTION
**Pull request by ohaim1008**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/27365

## Description
redcanary-get-detection is causing timeout because of endless loop.

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/RedCanary/Integrations/RedCanary/RedCanary.py [hunk 1]
```diff
@@ -174,17 +174,21 @@ def get_full_timeline(detection_id, per_page=100):
...
+
     while not done:
         res = http_get('/detections/{}/timeline'.format(detection_id),
                        params={
                            'page': page,
                            'per_page': per_page,
         })
-
-        if len(res['data']) == 0:
+        current_data = res['data']
```
Comment (bakatzir) [inline Packs/RedCanary/Integrations/RedCanary/RedCanary.py:185]:
```suggestion
        current_data = res.get('data')
```

File: Packs/RedCanary/Integrations/RedCanary/RedCanary.py [hunk 2]
```diff
@@ -174,17 +174,21 @@ def get_full_timeline(detection_id, per_page=100):
...
+        if len(current_data) == 0 or current_data == last_data:
```
Comment (bakatzir) [inline Packs/RedCanary/Integrations/RedCanary/RedCanary.py:186]:
```suggestion
        if not current_data or len(current_data) == 0 or current_data == last_data:
```

File: Packs/RedCanary/Integrations/RedCanary/RedCanary_test.py [hunk 3]
```diff
@@ -179,3 +199,24 @@ def test_fetch_multiple_times_with_new_incidents(mocker):
     # only one incidents is being created out of the 2 that were fetched
     assert len(incidents) == 1
     assert last_run["time"] == latest_time_of_occurrence_of_incidents2
+
+
+def test_def_get_full_timeline(mocker):
+    """Unit test
+    Given
+    - raw response of the http request
```
Comment (bakatzir) [inline Packs/RedCanary/Integrations/RedCanary/RedCanary_test.py:207]: what is the difference in between result1 and 2? please explain in the docstring.

File: Packs/RedCanary/ReleaseNotes/1_0_4.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Red Canary
+- Fixed an issue where **redcanary-get-detection** command cause timeout.
```
Comment (bakatzir) [inline Packs/RedCanary/ReleaseNotes/1_0_4.md:4]:
```suggestion
- Fixed an issue where **redcanary-get-detection** command caused timeout.
```

File: Packs/RedCanary/Integrations/RedCanary/RedCanary.py (hunk 2 above)
Comment (bakatzir) [reply Packs/RedCanary/Integrations/RedCanary/RedCanary.py:186]: also explain in a comment here what it means that these conditions are met.

**Review by bakatzir — APPROVED** [review]

File: Packs/RedCanary/ReleaseNotes/1_0_4.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Red Canary
+- Fixed an issue where **redcanary-get-detection** command caused timeout.
```
Comment (amshamah419) [inline Packs/RedCanary/ReleaseNotes/1_0_4.md:4]:
```suggestion
- Fixed an issue where the **redcanary-get-detection** command would cause a timeout.
```

**Pull request merged**